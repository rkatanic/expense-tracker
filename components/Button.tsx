interface Props {
  text: string;
  type?: "button" | "submit";
  onClick?: () => void;
  fullWidth?: boolean;
}

const Button = ({ text, type = "button", onClick, fullWidth }: Props) => {
  return (
    <button
      className={`btn-primary ${fullWidth ? "btn-full-width" : ""}`}
      type={type}
      onClick={onClick}
    >
      {text}
    </button>
  );
};

export default Button;
