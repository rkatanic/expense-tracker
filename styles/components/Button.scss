.btn {
    &-primary {
        border-radius: 10rem;
        background-color: var(--color-primary-500);
        border: none;
        font-size: 0.875rem;
        font-weight: 600;
        padding: 0.5rem 2rem;

        &:hover {
            cursor: pointer;
            background-color: var(--color-primary-600);
        }
    }

    &-full-width {
        width: 100%;
    }
}