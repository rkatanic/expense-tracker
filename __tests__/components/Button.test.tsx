import { fireEvent, render } from "@testing-library/react";
import Button from "../../components/Button";

describe("Button", (): void => {
  it("should render", (): void => {
    const { baseElement } = render(
      <Button text="button" onClick={jest.fn()} fullWidth />
    );

    expect(baseElement).toMatchSnapshot();
  });

  it("should call onClick handler", (): void => {
    const mockOnClick = jest.fn();
    const { getByText } = render(
      <Button text="button" onClick={mockOnClick} />
    );

    fireEvent.click(getByText("button"));

    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });
});
