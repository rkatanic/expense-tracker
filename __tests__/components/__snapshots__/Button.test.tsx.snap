// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
<body>
  <div>
    <button
      class="btn-primary btn-full-width"
      type="button"
    >
      button
    </button>
  </div>
</body>
`;
